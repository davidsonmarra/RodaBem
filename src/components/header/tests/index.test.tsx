import React from 'react';
import {create} from 'react-test-renderer';
import Header, {Props} from '..';
import {Text} from '@components';
import {TouchableOpacity} from 'react-native';

const mockText = 'Snapshot test!';
const createComponent = ({text = mockText, onBack}: Props) =>
  create(<Header text={text} onBack={onBack} />);

const mockGoBack = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({goBack: mockGoBack}),
}));

describe('Button', () => {
  it('renders correctly', () => {
    const tree = createComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders text correctly', () => {
    const instance = createComponent({}).root;
    expect(instance.findByType(Text)).toBeTruthy();
  });

  it('should call goBack', () => {
    const instance = createComponent({onBack: mockGoBack}).root;
    instance.props.onBack();
    expect(mockGoBack).toHaveBeenCalled();
  });

  it('should call onBack when press back icon', () => {
    const instance = createComponent({onBack: mockGoBack}).root;
    const backIcon = instance.findAllByType(TouchableOpacity)[0];
    backIcon.props.onPress();
    expect(mockGoBack).toHaveBeenCalled();
  });
});
