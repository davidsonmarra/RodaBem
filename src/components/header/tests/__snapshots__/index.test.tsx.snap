// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "paddingHorizontal": 12,
      "paddingVertical": 8,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "width": 24,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="#E30B0B"
      focusable={false}
      height="24"
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={960}
      vbWidth={960}
      width="24"
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4293069579,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M359-242L120-481l239-239 43 43-166 166h604v60H236l166 166-43 43z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "width": 12,
      }
    }
  />
  <Text
    style={
      [
        {
          "color": "#222222",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        {
          "flex": 1,
          "textAlign": "center",
        },
      ]
    }
  >
    Snapshot test!
  </Text>
  <View
    style={
      {
        "width": 12,
      }
    }
  />
  <View
    style={
      {
        "width": 24,
      }
    }
  />
</View>
`;
